File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/base/visual/PseudoVisualData.java
package base.visual;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Adrin on 6/4/2016.
 */
public class PseudoVisualData implements IDynamicPropertiesObject {

    private Map<String, Object> propertiesMap;

    private Map<String, Boolean> staticAnimatingPropertiesMap = new HashMap<String, Boolean>();

    public PseudoVisualData() {
        super();

        propertiesMap = new HashMap<String, Object>();
    }

    @Override
    public Object getPropertyValue(String property) {
        return propertiesMap.get(property);
    }

    @Override
    public void setPropertyValue(String property, Object value) {
        propertiesMap.put(property, value);
    }

    public Boolean getStaticAnimatingPropertyStatus(String property) {
        return staticAnimatingPropertiesMap.get(property);
    }

    public void putStaticAnimatingPropertyStatus(String property, Boolean status) {
        staticAnimatingPropertiesMap.put(property, status);
    }

}

File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/graphics/visual/drawable/PseudoVisualData.cpp
#include "PseudoVisualData.h"
#include <base/exceptions/LogicException.h>

PseudoVisualData::PseudoVisualData() {
    //void
}

std::shared_ptr<base::IWrappedValue> PseudoVisualData::getPropertyValue(std::string property) {
    return propertiesMap.getDirect(property);
}

void PseudoVisualData::setPropertyValue(std::string property, std::shared_ptr<base::IWrappedValue> value) {
    propertiesMap.putReference(property, value);
}

bool PseudoVisualData::isInit_StaticAnimatingPropertyStatus(std::string& property) {
    if(staticAnimatingPropertiesMap.getDirect(property, -1) != -1) {
        return true;
    }

    return false;
}

bool PseudoVisualData::isTrue_StaticAnimatingPropertyStatus(std::string& property) {
    int v = staticAnimatingPropertiesMap.getDirect(property, -1);

    if(v == 1) {
        return true;
    }else if(v == 0) {
        return false;
    }

    throw LogicException(LOC);
}

void PseudoVisualData::putStaticAnimatingPropertyStatus(std::string& property, int status) {
    staticAnimatingPropertiesMap.putDirect(property, status);
}

PseudoVisualData::~PseudoVisualData() {
    //void
}
